src/main/java/mods/eln/integration/waila/GhostNodeWailaProvider.kt

```diff
@@ -0,0 +1,82 @@
+package mods.eln.integration.waila
+
+import cpw.mods.fml.common.Optional
+import mcp.mobius.waila.api.IWailaConfigHandler
+import mcp.mobius.waila.api.IWailaDataAccessor
+import mcp.mobius.waila.api.IWailaDataProvider
+import mods.eln.Eln
+import mods.eln.misc.Coordonate
+import mods.eln.packets.GhostNodeWailaRequestPacket
+import net.minecraft.client.Minecraft
+import net.minecraft.entity.player.EntityPlayerMP
+import net.minecraft.item.ItemStack
+import net.minecraft.nbt.NBTTagCompound
+import net.minecraft.tileentity.TileEntity
+import net.minecraft.world.World
+
+@Optional.Interface(iface = "mcp.mobius.waila.api.IWailaDataProvider", modid = "Waila")
+class GhostNodeWailaProvider(private val transparentNodeHandler: TransparentNodeHandler): IWailaDataProvider {
+    companion object {
+        var updateTime = Minecraft.getSystemTime()
+    }
+
+    private fun getGhostData(coord: Coordonate): GhostNodeWailaData? {
+        var ghostData: GhostNodeWailaData? = null
+
+        if (Minecraft.getSystemTime() - GhostNodeWailaProvider.updateTime > 2000) {
+            Eln.elnNetwork.sendToServer(GhostNodeWailaRequestPacket(coord))
+            GhostNodeWailaProvider.updateTime = Minecraft.getSystemTime()
+        }
+        try {
+            ghostData = WailaCache.ghostNodes.get(coord)
+        } catch(e: Exception) {}
```

Comment:
The cache is not really adapted to our needs, so I guess the actual solution is the simplest. I do not want to save futures in a list in order to wait to call them when the response from the server has arrived nor do I want to spawn a thread for each pending message. Its a hack to use something a little different than it's intention, but it works.